modules/local/process/generate_proteins_from_vcf.nf

```diff
@@ -0,0 +1,41 @@
+// Import generic module functions
+include { initOptions; saveFiles } from './functions'
+
+params.options = [:]
+options    = initOptions(params.options)
+
+def VERSIONFRED2 = '2.0.6'
+def VERSIONMHCNUGGETS = '2.3.2'
+
+//TODO: combine in a subflow --> when needs to be removed
+process GENERATE_PROTEINS_FROM_VCF {
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:'.', publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::fred2=2.0.6 bioconda::mhcflurry=1.4.3 bioconda::mhcnuggets=2.3.2" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/mulled-v2-689ae0756dd82c61400782baaa8a7a1c2289930d:a9e10ca22d4cbcabf6b54f0fb5d766ea16bb171e-0"
+    } else {
+        container "quay.io/biocontainers/mulled-v2-689ae0756dd82c61400782baaa8a7a1c2289930d:a9e10ca22d4cbcabf6b54f0fb5d766ea16bb171e-0"
+    }
+
+    input:
+        tuple val(Sample), val(id), file(fasta_file), val(d), file(vcf_file)
```

Comment:
```suggestion
        tuple val(meta), path(fasta_file), path(vcf_file)
```

Reply:
For DSL2 processes, we try to add all the metadata that is needed to run the analysis in a Groovy `Linkedhasmap` object. This object would contain all the necessary metadata info. E.g. in your case it could be accessed like: `meta.id`, `meta.Sample`, `meta.d`. You can check out as example any module under the `nf-core/modules` repository, and all of them should contain this `meta` map. Here there is an example on how to create the `meta` object from the samplesheet information: https://github.com/nf-core/rnaseq/blob/master/subworkflows/local/input_check.nf

Reply:
For nextflow DSL2, the syntax `path` is rather use than `file`, even though `file` will not be eliminated but `path` is recommended instead.

Reply:
Included the meta object